lib/Sema/ConstraintSystem.cpp

```diff
@@ -2763,14 +2763,21 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
   if (!diff.overloads.empty())
     return false;
 
-  if (!llvm::all_of(solutions, [](const Solution &solution) -> bool {
+  bool noFixes = llvm::all_of(solutions, [](const Solution &solution) -> bool {
+    return solution.Fixes.empty();
```

Comment:
There could be N fixes but all of them might be warnings, so it might be better to check for `SK_Fix` score being `0`. Also we can do it in one go together with all mismatches, right?

Reply:
> There could be N fixes but all of them might be warnings, so it might be better to check for SK_Fix score being 0.

Sure!

> Also we can do it in one go together with all mismatches, right?

Not sure, but either way I think this is cleaner to separate the logics, and since it's not performance sensitive... maybe is ok